controllers/sts.go

```diff
@@ -234,33 +234,18 @@ func (r *DirectoryServiceReconciler) createDSStatefulSet(ctx context.Context, ds
 			Replicas:    ds.Spec.Replicas,
 			Template: v1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
-					Labels: createLabels(ds.Name, map[string]string{
-						"affinity": "directory", // for anti-affinity
-					}),
+					Labels: createLabels(ds.Name, nil),
 				},
 				Spec: v1.PodSpec{
-					// We use anti affinity to spread the pods out over host node
-					Affinity: &v1.Affinity{
-						// NodeAffinity:    &v1.NodeAffinity{},
-						// PodAffinity:     &v1.PodAffinity{},
-						PodAntiAffinity: &v1.PodAntiAffinity{
-							//RequiredDuringSchedulingIgnoredDuringExecution:  []v1.PodAffinityTerm{},
-							PreferredDuringSchedulingIgnoredDuringExecution: []v1.WeightedPodAffinityTerm{
-								{
-									Weight: 100,
-									PodAffinityTerm: v1.PodAffinityTerm{
-										LabelSelector: &metav1.LabelSelector{
-											MatchLabels: map[string]string{},
-											MatchExpressions: []metav1.LabelSelectorRequirement{
-												{
-													Key:      "affinity",
-													Operator: "In",
-													Values:   []string{"directory"},
-												},
-											},
-										},
-										TopologyKey: "kubernetes.io/hostname",
-									},
+					// Spread the DS pods across zones if possible. If not possible, schedule anyways
+					TopologySpreadConstraints: []v1.TopologySpreadConstraint{
+						{
+							MaxSkew:           1,
+							TopologyKey:       "zone",
```

Comment:
```suggestion
							TopologyKey:       ""topology.kubernetes.io/zone"",
```
shouldn't this be `topology.kubernetes.io/zone`. I know the docs mention "zone" but our nodes are not tagged with that label. Also, later on in the docs they have `topology.kubernetes.io/zone` which is the default label.


I would suggest to give this a go in AWS and Azure just in case, or at least verify they tag the nodes with the correct labels.